File: my-code/434/main.py
# -*- coding: UTF-8 -*-
# @Time     : 2021/08/20 23:18
# @Author   : Ranshi
# @File     : main.py


class Solution:
    def countSegments(self, s: str) -> int:
        return len([item for item in s.split(" ") if item])


if __name__ == "__main__":
    print(Solution().countSegments(""))

File: my-code/434/main.go
package main

import "fmt"

// @Time     : 2021/10/07 11:34
// @Author   : Ranshi
// @File     : main.go

func countSegments(s string) (ans int) {
	for i, ch := range s {
		if (i == 0 || s[i-1] == ' ') && ch != ' ' {
			ans++
		}
	}
	return
}

func main() {
	fmt.Println(countSegments("Hello, my name is John"))
}
